File: frontend/src/types/building.ts
// Mirrors app/schemas/building.py exactly.

export interface Building {
  id: number;
  name: string;
  address: string | null;
}

export interface BuildingCreateInput {
  name: string;
  address?: string | null;
}

export interface BuildingUpdateInput {
  name?: string;
  address?: string | null;
}

File: backend/app/models/building.py
from sqlalchemy import Integer, String
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.db.base import Base


class Building(Base):
    __tablename__ = "buildings"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, index=True)
    name: Mapped[str] = mapped_column(String, nullable=False)
    address: Mapped[str | None] = mapped_column(String, nullable=True)

    floors = relationship(
        "Floor",
        back_populates="building",
        cascade="all, delete-orphan",
        passive_deletes=True,
    )
